components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/CertificateMgtUtils.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * you may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.apimgt.impl.utils;
+
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.apimgt.impl.certificatemgt.ResponseCode;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.UnsupportedEncodingException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+
+/**
+ * This class holds the utility methods for certificate management.
+ */
+public class CertificateMgtUtils {
+    private static Log log = LogFactory.getLog(CertificateMgtUtils.class);
+    private static String TRUST_STORE_PASSWORD = System.getProperty("javax.net.ssl.trustStorePassword");
+    private static String TRUST_STORE = System.getProperty("javax.net.ssl.trustStore");
+
+    /**
+     * This method generates a certificate from a base64 encoded certificate string and add to the configured trust
+     * store.
+     *
+     * @param base64Cert : The base 64 encoded string of the server certificate.
+     * @param alias      : The alias for the certificate.
+     * @return : ResponseCode which matches the execution result.
+     *
+     * Response Codes.
+     * SUCCESS : If certificate added successfully.
+     * INTERNAL_SERVER_ERROR : If any internal error occurred
+     * ALIAS_EXISTS_IN_TRUST_STORE : If the alias exists in trust store.
+     * CERTIFICATE_EXPIRED : If the given certificate is expired.
+     */
+    public ResponseCode addCertificateToTrustStore(String base64Cert, String alias) {
+        boolean isCertExists = false;
+        boolean expired = false;
+        try {
+            //Decode base64 encoded certificate.
+            byte[] cert = (Base64.decodeBase64(base64Cert.getBytes("UTF-8")));
```

Comment:
what if we dont specify UTF-8 here. and let system configured char system to use. also we can move this to a constant if we can't do it